### PR TITLE
global: add class based registry to extend idutils schemes

### PR DESCRIPTION
**Pull request by zzacharo**
closes https://github.com/inveniosoftware/idutils/issues/105

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -37,7 +37,6 @@ install_requires =
 
 [options.extras_require]
 tests =
-    invenio-app>=1.4.0
     pytest-black-ng>=0.4.0
     pytest-cache>=1.0
     pytest-runner>=2.6.2
```
Comment (utnapischtim) [inline setup.cfg:42]: should the section `[options.entry_points]` be removed?

File: idutils/detectors.py [hunk 2]
```diff
@@ -14,70 +14,18 @@
...
-    ("uniprot", validators.is_uniprot),
-    ("refseq", validators.is_refseq),
-    ("genome", validators.is_genome),
-    ("geo", validators.is_geo),
-    ("arrayexpress_array", validators.is_arrayexpress_array),
-    ("arrayexpress_experiment", validators.is_arrayexpress_experiment),
-    ("swh", validators.is_swh),
-    ("viaf", validators.is_viaf),
-]
+IDUTILS_PID_SCHEMES = _IDUTILS_PID_SCHEMES
```
Comment (zzacharo) [inline idutils/detectors.py:21]: Kept for backwards compatibility when importing.

File: idutils/detectors.py [hunk 3]
```diff
@@ -14,70 +14,18 @@
...
-    (
-        "issn",
-        [
-            "gnd",
-            "viaf",
-        ],
-    ),
-    ("pmid", ["viaf"]),
-]
+IDUTILS_SCHEME_FILTER = _IDUTILS_SCHEME_FILTER
```
Comment (zzacharo) [inline idutils/detectors.py:28]: Kept for backwards compatibility when importing.

File: setup.cfg (hunk 1 above)
Comment (zzacharo) [reply setup.cfg:42]: Yes! Nice catch :)

**Review by tmorrell — APPROVED** [review]
I've tested and this doesn't break my usage. It would be useful to have some documentation on how the scheme extension works. This is one repository where maybe readthedocs would be a good place, just in https://github.com/inveniosoftware/idutils/blob/master/docs/index.rst. It will need https://github.com/inveniosoftware/idutils/pull/107 merged at a minimum to render.

**Comment by zzacharo** [comment]
> I've tested and this doesn't break my usage. It would be useful to have some documentation on how the scheme extension works. This is one repository where maybe readthedocs would be a good place, just in https://github.com/inveniosoftware/idutils/blob/master/docs/index.rst. It will need #107 merged at a minimum to render.

Hey, thanks for testing it! I have added some documentation in https://github.com/inveniosoftware/idutils/blob/master/idutils/__init__.py#L19 but I am happy to add it also as part of the readthedocs too.

**Pull request merged**